packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,2896 @@
+  /// object should be provided.
+  ///
+  /// {@macro flutter.material.menu_bar.shortcuts_note}
+  final List<Widget> children;
+
+  @override
+  State<MenuBar> createState() => _MenuBarState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DiagnosticsProperty<MenuController?>('controller', controller, defaultValue: null));
+    properties.add(
+        DiagnosticsProperty<MaterialStateProperty<Color?>>('backgroundColor', backgroundColor, defaultValue: null));
+    properties.add(DoubleProperty('minimumHeight', minimumHeight, defaultValue: null));
+    properties.add(DiagnosticsProperty<EdgeInsetsDirectional?>('padding', padding, defaultValue: null));
+    properties.add(DiagnosticsProperty<MaterialStateProperty<double?>?>('elevation', elevation, defaultValue: null));
+  }
+}
+
+class _MenuBarState extends State<MenuBar> with DiagnosticableTreeMixin {
+  MenuController? _internalController;
+  MenuController get controller {
+    return widget.controller ?? (_internalController ??= MenuController());
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    assert(() {
+      controller._root.menuScopeNode.debugLabel = 'MenuBar';
+      return true;
+    }());
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+  }
+
+  @override
+  void didUpdateWidget(covariant MenuBar oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.controller != null) {
+      _internalController?.dispose();
+      _internalController = null;
+    }
+  }
+
+  @override
+  void dispose() {
+    _internalController?.dispose();
+    _internalController = null;
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    assert(debugCheckHasOverlay(context));
+    final MenuThemeData menuTheme = MenuTheme.of(context);
+    return AnimatedBuilder(
+      animation: controller,
+      builder: (BuildContext context, Widget? ignoredChild) {
+        return ExcludeFocus(
+          excluding: !controller.menuIsOpen,
+          child: TapRegion(
+            groupId: controller,
+            onTapOutside: (PointerDownEvent event) {
+              controller.closeAll();
+            },
+            child: _MenuControllerMarker(
+              controller: controller,
+              child: FocusScope(
+                node: controller._root.menuScopeNode,
+                child: Actions(
+                  actions: <Type, Action<Intent>>{
+                    DirectionalFocusIntent: _MenuDirectionalFocusAction(controller: controller),
+                    DismissIntent: _MenuDismissAction(controller: controller),
+                  },
+                  child: Shortcuts(
+                    shortcuts: _kMenuTraversalShortcuts,
+                    child: _MenuPanel(
+                      elevation: (widget.elevation ?? menuTheme.barElevation ?? _TokenDefaultsM3(context).barElevation)
```

Comment:
Default resolution like this is OK now because we're not resolving the MaterialStateProperties against any states. If we were then it would be necessary resolve each value:
```
widget.elevation?.resolve(states) ?? menuTheme.barElevation?.resolves(states) ?? _TokenDefaultsM3(context).barElevation!.resolve(states);
```
I'm only mentioning it because a future version of the code might introduce states and not get this right.